src/databricks/labs/remorph/transpiler/transpile_engine.py

```diff
@@ -7,10 +9,44 @@
 
 class TranspileEngine(abc.ABC):
 
+    @classmethod
+    def load_engine(cls, transpiler: Path) -> TranspileEngine:
```

Comment:
Why can't the load_engine be a abstract and implementation is pushed to the actual engine itself ?

Reply:
Not sure I understand your comment ? This class method is about instantiating the engine such that other methods can be abstract...

Reply:
Why not but what problem would that solve ?
In the end state, the SQLGlot engine will go away (becoming a pluggable transpiler), so this code will become simpler.